From 210422d554fc8fe7e013fc48af2eafb7a1690293 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Fri, 6 Mar 2026 14:51:34 +0100
Subject: [PATCH] mmc: fsl_esdhc: call cyclic schedule() during dma transfers

Large-size transfers from especially microsd cards can take a long time
and easily exceed a typical 30s watchdog timer, leading to unexpected
reset e.g. during load commands.

Add call to WATCHDOG_RESET() function during dma transfer.

This fixes in particular an issue on LX2160 when loading a large disk
image from microSD and writing to eMMC would be interrupted because
expired watchdog reset the system after 60 seconds.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 drivers/mmc/fsl_esdhc.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/mmc/fsl_esdhc.c b/drivers/mmc/fsl_esdhc.c
index b3c71c86950..9809354e1e8 100644
--- a/drivers/mmc/fsl_esdhc.c
+++ b/drivers/mmc/fsl_esdhc.c
@@ -29,6 +29,7 @@
 #include <linux/delay.h>
 #include <linux/dma-mapping.h>
 #include <sdhci.h>
+#include <watchdog.h>
 
 DECLARE_GLOBAL_DATA_PTR;
 
@@ -465,6 +466,9 @@ static int esdhc_send_cmd_common(struct fsl_esdhc_priv *priv, struct mmc *mmc,
 					err = -ECOMM;
 					goto out;
 				}
+
+				/* service watchdog */
+				WATCHDOG_RESET();
 			} while ((irqstat & flags) != flags);
 
 			/*
-- 
2.51.0

